### PR TITLE
Add User Input End Time

### PR DESCRIPTION
**Pull request by mengyaoyang11**
Problem:
Custom Time Search new UI has end time set either as default now or as customer input. 

Solution:
Update computeTimeRangeInternal function to make sure no error return when endTimeVal and lookBackVal are given

Note:
1) keep startTimeVal as a parameter for now even though its not exposed to both current/new UI, will totally remove startTimeVal once we are sure its not used anywhere in new feature.
2) Keep the input validation for now even though the new feature ensures lookBackVal will not be empty.

### CODE REVIEW

File: pkg/sloop/queries/timerange.go [hunk 1]
```diff
@@ -62,18 +62,18 @@ func computeTimeRangeInternal(params url.Values, endOfTime time.Time, maxLookBac
 		}
 	}
 
+	// always use customer input as computedEnd
+	computedStart, computedEnd, err = getTimeRangeFromStartEnd(startTimeVal, endTimeVal)
```
Comment (nurland) [inline pkg/sloop/queries/timerange.go:66]: @mengyaoyang11 we are breaking the existing behavior - if lookBack is passed then startTime and endTime will be empty and getTimeRangeFromStartEnd will return error.

**Review by nurland — CHANGES_REQUESTED** [review]

File: pkg/sloop/queries/timerange_test.go [hunk 2]
```diff
@@ -55,7 +59,8 @@ func Test_timeRangeTable(t *testing.T) {
 		{"", "abc", "123", true, time.Time{}, time.Time{}},
 		{"", "123", "abc", true, time.Time{}, time.Time{}},
 	}
-	for _, thisRange := range rangeTests {
+	for i, thisRange := range rangeTests {
+		fmt.Print(i)
```
Comment (nurland) [inline pkg/sloop/queries/timerange_test.go:63]: why do we need this? can we remove it?

Comment (mengyaoyang11) [reply pkg/sloop/queries/timerange_test.go:63]: It helps especially when test case failed. With the number printed out you can easily target which test is failed. Of course we can remove it does not make sense.

File: pkg/sloop/queries/timerange.go (hunk 1 above)
Comment (mengyaoyang11) [reply pkg/sloop/queries/timerange.go]: updated

File: pkg/sloop/queries/timerange_test.go (hunk 2 above)
Comment (nurland) [reply pkg/sloop/queries/timerange_test.go:63]: Can we just use t.Run arg for that? E.g. we can update the line below to be something like this:

t.Run(fmt.Sprintf("Test id: %v, range: %+v", i, thisRange)

Comment (mengyaoyang11) [reply pkg/sloop/queries/timerange_test.go]: updated

File: pkg/sloop/queries/timerange_test.go [hunk 3]
```diff
@@ -55,8 +59,8 @@ func Test_timeRangeTable(t *testing.T) {
 		{"", "abc", "123", true, time.Time{}, time.Time{}},
 		{"", "123", "abc", true, time.Time{}, time.Time{}},
 	}
-	for _, thisRange := range rangeTests {
-		t.Run(fmt.Sprintf("%+v", thisRange), func(t *testing.T) {
+	for i, thisRange := range rangeTests {
+		t.Run(fmt.Sprintf("Test id: %v, thisRange: %v", i, thisRange), func(t *testing.T){
```
Comment (nurland) [inline pkg/sloop/queries/timerange_test.go:63]: nit: just curious why did we change `%+v` to `%v`?

Comment (mengyaoyang11) [reply pkg/sloop/queries/timerange_test.go]: you are correct, change back to `%+v` as plus flag adds field names

**Review by nurland — APPROVED** [review]

**Pull request merged**